### DISCUSSION TITLE
Filter.removeUnavailable удаляет все треки

### DISCUSSION THREAD

**Discussion by amd64fox** [Chimildic/goofy#180] [Помощь с ошибкой] [answered]
У меня почему-то `Filter.removeUnavailable`  удаляет все подряд в массивах под ноль, ошибок не выдает. 
Временно отключил этот фильтр где использовался, судя по выдачи, началось это дней 5 назад.
Есть у кого-то похожая с ним проблема ?

**Comment by Chimildic** [comment, marked as answer]
[removeUnavailable](https://chimildic.github.io/goofy/#/reference/filter?id=removeunavailable) вторым аргументом принимает `market` (страна/рынок), по которому определяется доступность треков. Значение по умолчанию `RU`.  После ухода Spotify 11 апреля, в метаданных от API нет `RU`. Поэтому фильтр срезает все. 

Если в [вашем профиле](https://www.spotify.com/account/profile/) по прежнему Россия, то фильтр бесполезен. Если страна менялась, то укажите ее вторым аргументом в соответствии с [двухбуквенным сокращением](https://www.artlebedev.ru/country-list/).

**Reply by amd64fox** [reply]
Да действительно проблема в стране была, поменял её в конфиге и фильтр снова заработал. 
Возможно тогда лучше убрать временно из конфига по умолчанию `'LOCALE', 'RU'`

**Reply by Chimildic** [reply]
У свойства `LOCALE` из файла _config_ другая задача. Известны случае, когда треки на кириллице приходили транскрипцией. Поэтому введена локаль. `market` у [removeUnavailable](https://chimildic.github.io/goofy/#/reference/filter?id=removeunavailable) делает другое.

**Reply by amd64fox** [reply]
Тогда убрать по умолчанию из кода функции [removeUnavailable](https://chimildic.github.io/goofy/#/reference/filter?id=removeunavailable)
